app/registries/foreman/setting_manager.rb

```diff
@@ -0,0 +1,74 @@
+module Foreman
+  class SettingManager
+    class << self
+      # Holds all the setting definitions for application
+      def settings
+        @settings ||= {}
+      end
+
+      # Holds all setting categories with their labels
+      def categories
+        @categories ||= { 'general' => N_('General') }
+      end
+
+      def map(context_name, &block)
+        mapper = Mapper.new(self, context_name.to_sym)
+        if block_given?
+          mapper.instance_eval(&block)
+        else
+          mapper
+        end
+      end
```

Comment:
I would think that `SettingsManager does an `instance_eval` on itself and defines a `category` method that evaluates on a `CategoryManager`. The `CategoryManager` would only support a single category. I think that keeps it simple.

The `SettingsManager` could also implement a `setting` method to define a setting in the `general` category.

Thoughts on this?

Reply:
I've implemented this bit differently keeping the main idea (I believe).
We now have `SettingManager`, that only keeps values and provide only method `define`, which creates instance of private class `SettingMapper` and `instance_eval` on that, which then only provides method `category`, that does the same only providing the final method `setting`. The code is indeed cleaner :+1: 
Only deviation is, that category is now mandatory, what is I believe better design to be explicit about category, especially when we probably don't want to encourage abusing category `general`.

Reply:
:+1: on forcing plugin authors to choose a category.